lib/src/main/java/graphql/nadel/Nadel.kt

```diff
@@ -48,7 +55,24 @@ class Nadel private constructor(
     private val instrumentation: NadelInstrumentation,
     private val preparsedDocumentProvider: PreparsedDocumentProvider,
 ) {
-    fun execute(nadelExecutionInput: NadelExecutionInput): CompletableFuture<ExecutionResult> {
+    private val coroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.Default)
+
+    @Deprecated(message = "Use executeAsync", replaceWith = ReplaceWith("this.executeAsync(executionInput)"))
+    @JvmName("execute") // For binary compat
+    fun executeJava(executionInput: NadelExecutionInput): CompletableFuture<ExecutionResult> {
+        return coroutineScope.future {
+            execute(executionInput)
+        }
+    }
+
+    fun executeAsync(executionInput: NadelExecutionInput): CompletableFuture<ExecutionResult> {
+        return coroutineScope.future {
+            execute(executionInput)
+        }
+    }
+
+    @JvmName("executeSuspending")
```

Comment:
Should this really be part of the API? I can't think of reasons why clients would call `execute` in a blocking way.

Also, is `executeSuspending` conveying the right meaning? What does "suspending" mean in Java code? Perhaps `executeBlocking` would be a better name.

Reply:
Well this isn't blocking, this is the suspending function in Kotlin. In a Kotlin codebase, this IS the entry point. The other functions are for Java compat.

Suspending functions can ONLY be invoked from Kotlin code. Java code can't invoke suspending functions.

1.  We had an `execute` function with the same name.
2. So we "renamed" that to `executeJava` in Kotlin, but Java code will see this as `execute` due to the `@JvmName` annotation.

I think the `@JvmName` is actually a bit redundant because the Kotlin compiler will change the function signature anyway, but I decided to keep it in to make it more explicit that we've kind of swapped these functions around.

Can explain this more at our sync.

Reply:
Right - yeah. I was focused on the return type and missed the `suspend` keyword. Makes sense.
Should we get rid of the `@JvmName`, this should make the method essentially inaccessible from Java right, and just add some comments?